### PR TITLE
Middleware solution and new endpoints

### PR DESCRIPTION
**Pull request by nisalikularatne**

### CODE REVIEW

File: src/authenticator/v1/middleware/authorization.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const Authorization = require('../authorization/index');
+let t;
```
Comment (daniakabani) [inline src/authenticator/v1/middleware/authorization.js:2]: what's t?

File: src/authenticator/v1/middleware/authorization.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    }
+    req.authorize = await authorize;
+    next();
+
+
+};
+const authorize = async function (object) {
+    const authorization = new Authorization().setToken(token);
+    await authorization.authorize(object);
+};
```
Comment (Ali-Dalal) [inline src/authenticator/v1/middleware/authorization.js:15]: don't define global singleton token, it is very bad practice in a multithreaded environment.

We've done something similar in Aggregator project, check the code there, please.

**Review by daniakabani — APPROVED** [review]
approved

File: test/helpers/index.js [hunk 3]
```diff
@@ -72,14 +72,22 @@ exports.generateToken = async () => {
...
         if (type === 'principal') {
-            let response = await new Principal().withAuth().create(`principal_test_${timeStamp}`, '111');
+            let response = await new Principal().withAuth().create({
+                name: `principal_test_${timeStamp}`,
+                dataset_id: '111'
+            });
             return response.id;
         }
     } else {
+        console.log('check here');
```
Comment (Ali-Dalal) [inline test/helpers/index.js:90]: console.log?

**Review by Ali-Dalal — CHANGES_REQUESTED** [review]
some minor changes are required. after, the PR ready to go

File: src/authenticator/v1/middleware/authorization.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+const authorize = function (auth_token, serviceNameMain) {
+    let service;
+    return async function ({ permissionName, resourceType, resourceID, datasetID, serviceName }) {
+        if (serviceName) {
+            service = serviceName;
+        } else {
+            service = serviceNameMain;
+        }
```
Comment (Ali-Dalal) [inline src/authenticator/v1/middleware/authorization.js:19]: can we replace this `if` with a logical operation?
```js
await authorization.authorize({
            serviceName: serviceName || serviceNameMain,
            permissionName,
            resourceType,
            resourceID,
            datasetID
        });
```

**Review by Ali-Dalal — CHANGES_REQUESTED** [review]
1 more fix

**Review by Ali-Dalal — APPROVED** [review]
✨

**Pull request merged**